[2023-04-23T15:51:54.369+0000] {taskinstance.py:1090} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: nfl_transformation_dag.task_pyspark scheduled__2023-04-11T00:00:00+00:00 [None]>
[2023-04-23T15:51:54.384+0000] {taskinstance.py:1090} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: nfl_transformation_dag.task_pyspark scheduled__2023-04-11T00:00:00+00:00 [None]>
[2023-04-23T15:51:54.386+0000] {taskinstance.py:1288} INFO - 
--------------------------------------------------------------------------------
[2023-04-23T15:51:54.387+0000] {taskinstance.py:1289} INFO - Starting attempt 37 of 39
[2023-04-23T15:51:54.388+0000] {taskinstance.py:1290} INFO - 
--------------------------------------------------------------------------------
[2023-04-23T15:51:54.409+0000] {taskinstance.py:1309} INFO - Executing <Task(_PythonDecoratedOperator): task_pyspark> on 2023-04-11 00:00:00+00:00
[2023-04-23T15:51:54.428+0000] {standard_task_runner.py:55} INFO - Started process 10748 to run task
[2023-04-23T15:51:54.431+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'nfl_transformation_dag', 'task_pyspark', 'scheduled__2023-04-11T00:00:00+00:00', '--job-id', '1211', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/data_ingestion.py', '--cfg-path', '/tmp/tmpem12zvp3']
[2023-04-23T15:51:54.434+0000] {standard_task_runner.py:83} INFO - Job 1211: Subtask task_pyspark
[2023-04-23T15:51:54.551+0000] {task_command.py:389} INFO - Running <TaskInstance: nfl_transformation_dag.task_pyspark scheduled__2023-04-11T00:00:00+00:00 [running]> on host 235fa867d994
[2023-04-23T15:51:54.637+0000] {taskinstance.py:1516} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=nfl_transformation_dag
AIRFLOW_CTX_TASK_ID=task_pyspark
AIRFLOW_CTX_EXECUTION_DATE=2023-04-11T00:00:00+00:00
AIRFLOW_CTX_TRY_NUMBER=37
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2023-04-11T00:00:00+00:00
[2023-04-23T15:51:54.642+0000] {subprocess.py:63} INFO - Tmp dir root location: 
 /tmp
[2023-04-23T15:51:54.650+0000] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', 'gcloud dataproc jobs submit pyspark                 --cluster=nfl-spark-cluster                 --region=europe-west6                 --jars=gs://spark-lib/bigquery/spark-bigquery-with-dependencies_2.12-0.30.0.jar                gs://nfl-data-lake_nfl-project-de/code/transform_pyspark.py                 --                     --year=2022                     --season_type=2 ']
[2023-04-23T15:51:54.691+0000] {subprocess.py:86} INFO - Output:
[2023-04-23T15:52:00.490+0000] {subprocess.py:93} INFO - Job [67adcb466dd24a74aea2386de837d393] submitted.
[2023-04-23T15:52:00.492+0000] {subprocess.py:93} INFO - Waiting for job output...
[2023-04-23T15:52:11.974+0000] {subprocess.py:93} INFO - 23/04/23 15:52:09 INFO org.apache.spark.SparkEnv: Registering MapOutputTracker
[2023-04-23T15:52:11.975+0000] {subprocess.py:93} INFO - 23/04/23 15:52:09 INFO org.apache.spark.SparkEnv: Registering BlockManagerMaster
[2023-04-23T15:52:11.976+0000] {subprocess.py:93} INFO - 23/04/23 15:52:09 INFO org.apache.spark.SparkEnv: Registering BlockManagerMasterHeartbeat
[2023-04-23T15:52:11.977+0000] {subprocess.py:93} INFO - 23/04/23 15:52:09 INFO org.apache.spark.SparkEnv: Registering OutputCommitCoordinator
[2023-04-23T15:52:11.978+0000] {subprocess.py:93} INFO - 23/04/23 15:52:09 INFO org.sparkproject.jetty.util.log: Logging initialized @4395ms to org.sparkproject.jetty.util.log.Slf4jLog
[2023-04-23T15:52:11.979+0000] {subprocess.py:93} INFO - 23/04/23 15:52:09 INFO org.sparkproject.jetty.server.Server: jetty-9.4.40.v20210413; built: 2021-04-13T20:42:42.668Z; git: b881a572662e1943a14ae12e7e1207989f218b74; jvm 1.8.0_322-b06
[2023-04-23T15:52:11.980+0000] {subprocess.py:93} INFO - 23/04/23 15:52:09 INFO org.sparkproject.jetty.server.Server: Started @4531ms
[2023-04-23T15:52:11.982+0000] {subprocess.py:93} INFO - 23/04/23 15:52:09 INFO org.sparkproject.jetty.server.AbstractConnector: Started ServerConnector@14d525b0{HTTP/1.1, (http/1.1)}{0.0.0.0:41611}
[2023-04-23T15:52:16.745+0000] {subprocess.py:93} INFO - 23/04/23 15:52:11 INFO com.google.cloud.hadoop.repackaged.gcs.com.google.cloud.hadoop.gcsio.GoogleCloudStorageImpl: Ignoring exception of type GoogleJsonResponseException; verified object already exists with desired state.
[2023-04-23T15:52:23.913+0000] {subprocess.py:93} INFO - 23/04/23 15:52:17 WARN org.apache.spark.sql.catalyst.util.package: Truncated the string representation of a plan since it was too large. This behavior can be adjusted by setting 'spark.sql.debug.maxToStringFields'.
[2023-04-23T15:52:23.914+0000] {subprocess.py:93} INFO - Cleaning the data for the teams_df started
[2023-04-23T15:52:23.915+0000] {subprocess.py:93} INFO - Cleaning the data for the teams_df is successful!
[2023-04-23T15:52:23.915+0000] {subprocess.py:93} INFO - 23/04/23 15:52:22 INFO org.apache.hadoop.io.compress.CodecPool: Got brand-new decompressor [.snappy]
[2023-04-23T15:52:24.867+0000] {local_task_job.py:276} WARNING - State of this instance has been externally set to None. Terminating instance.
[2023-04-23T15:52:24.878+0000] {process_utils.py:129} INFO - Sending Signals.SIGTERM to group 10748. PIDs of all processes in the group: [10749, 10748]
[2023-04-23T15:52:24.879+0000] {process_utils.py:84} INFO - Sending the signal Signals.SIGTERM to group 10748
[2023-04-23T15:52:24.881+0000] {taskinstance.py:1488} ERROR - Received SIGTERM. Terminating subprocesses.
[2023-04-23T15:52:24.893+0000] {taskinstance.py:1776} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/decorators/base.py", line 217, in execute
    return_value = super().execute(context)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 175, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 192, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/data_ingestion.py", line 231, in task_pyspark
    operator.execute(context={})
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/bash.py", line 187, in execute
    result = self.subprocess_hook.run_command(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/hooks/subprocess.py", line 91, in run_command
    for raw_line in iter(self.sub_process.stdout.readline, b""):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1490, in signal_handler
    raise AirflowException("Task received SIGTERM signal")
airflow.exceptions.AirflowException: Task received SIGTERM signal
[2023-04-23T15:52:24.913+0000] {taskinstance.py:1327} INFO - Marking task as UP_FOR_RETRY. dag_id=nfl_transformation_dag, task_id=task_pyspark, execution_date=20230411T000000, start_date=20230423T155154, end_date=20230423T155224
[2023-04-23T15:52:24.938+0000] {standard_task_runner.py:100} ERROR - Failed to execute job 1211 for task task_pyspark (Task received SIGTERM signal; 10748)
[2023-04-23T15:52:24.973+0000] {process_utils.py:79} INFO - Process psutil.Process(pid=10748, status='terminated', exitcode=1, started='15:51:53') (10748) terminated with exit code 1
[2023-04-23T15:52:29.260+0000] {process_utils.py:79} INFO - Process psutil.Process(pid=10749, status='terminated', started='15:51:53') (10749) terminated with exit code None
